cmd/evm/internal/t8ntool/transition.go

```diff
@@ -262,6 +262,25 @@ func Transition(ctx *cli.Context) error {
 			return NewError(ErrorConfig, errors.New("EIP-1559 config but missing 'currentBaseFee' in env section"))
 		}
 	}
+	// Sanity check pre-allocated EOF code to not panic in state transition.
+	if chainConfig.IsShanghai(big.NewInt(int64(prestate.Env.Number))) {
```

Comment:
I think t8n tool should also check that before the EOF fork, code in prestate doesn't start with `EF00`